aws/components/directory/state.ftl

```diff
@@ -0,0 +1,80 @@
+[#ftl]
+[#macro aws_directory_cf_state occurrence parent={} ]
+    [#local core = occurrence.Core]
+    [#local solution = occurrence.Configuration.Solution ]
+
+    [#local id = formatResourceId(AWS_DIRECTORY_RESOURCE_TYPE, core.Id) ]
+    [#local securityGroupId = formatDependentSecurityGroupId(id)]
+
+    [#local segmentSeedId = formatSegmentSeedId() ]
+    [#local segmentSeed = getExistingReference(segmentSeedId)]
+
+    [#local certificateObject = getCertificateObject(solution.Certificate!"")]
+    [#local certificateDomains = getCertificateDomains(certificateObject) ]
+    [#local primaryDomainObject = getCertificatePrimaryDomain(certificateObject) ]
+    [#local hostName = getHostName(certificateObject, occurrence) ]
+    [#local fqdn = formatDomainName(hostName, primaryDomainObject) ]
+
+    [#local dnsPorts = [ 
+        "dns-tcp", "dns-tcp",
+        "globalcatalog",
+        "kerebosauth88-tcp", "kerebosauth88-udp", "kerebosauth464-tcp", "kerebosauth464-udp",
+        "ldap-tcp", "ldap-udp", "ldaps",
+        "netlogin-tcp", "netlogin-udp", 
+        "ntp",
+        "rpc", "ephemeralrpctcp", "ephemeralrpcudp",
+        "rsync",
+        "smb-tcp", "smb-udp", 
+        "anyicmp"
+    ]]
```

Comment:
To remove duplication of quite a bit list,  you could put these as an attribute on the directory resource as Ports or something like that. Then in the setup routine you can just reference the resource details instead of having to duplicate the array. 
One niggly thing, these are only the ports when using AD not when you are using simple AD right? Do they both support stuff like RPC?

Reply:
You're correct RPC is not, I've left LDAPS in (even though it is recommended against) and RSYNC as I'm not sure on the replication used for SimpleAD